prow/tide/tide_test.go

```diff
@@ -216,38 +228,60 @@ func TestAccumulateBatch(t *testing.T) {
 		},
 	}
 	for _, test := range tests {
-		var pulls []PullRequest
-		for _, p := range test.pulls {
-			pr := PullRequest{
-				Number:     githubql.Int(p.number),
-				HeadRefOID: githubql.String(p.sha),
+		t.Run(test.name, func(t *testing.T) {
+			config.FakeInRepoConfig = test.fakeInRepoConfig
+			defer func() {
+				config.FakeInRepoConfig = nil
```

Comment:
Is this threadsafe in the context of parallel tests? Why do we need to mutate global state for mocking? Why not inject the dependency?

Reply:
Its not, which is not a problem as none of these tests are being executed in parallel. I can make `FakeInRepo` config a property of `config.Config` instead of a package-level var though.

Reply:
I think making it a file of the config struct instead of global would be nice. We can use the field annotation ``` `json:"-"` ``` to prevent users from specifying this in config since it is only for testing.

Reply:
Updated this